tempto-core/src/main/java/com/teradata/tempto/hadoop/FileSystemClient.java

```diff
@@ -45,15 +45,17 @@ InputStream getInputStream()
 
     void createDirectory(String path);
 
+    void deleteDirectory(String path);
```

Comment:
Please let keep the convention of having a single commit per each logical change. Also refactor commits should not change the code semantic, while commits which changes the semantic (add new stuff) should not have any refactor changes.

Having this in mind, I would keep move/rename of `HdfsClient` to `FileSystemClient` as separate commits, and then on top of them I would introduce new methods (or any new stuff) in subsequent commits.